File: scan-app/app/src/main/java/scan/listener/ScanListener.java
package scan.listener;

import android.content.Context;

import java.util.List;

import scan.AppModel;

/**
 * by y on 2017/2/15
 */

public interface ScanListener {
    void onScanStart();

    void onScanSuccess(List<AppModel> data);

    void onScanError(Throwable e);

    Context getScanContext();
}

File: ScanApp/app/src/main/java/scan/listener/ScanListener.kt
package scan.listener

import android.content.Context

import scan.AppModel

/**
 * by y on 2017/2/15
 */

interface ScanListener {
    val scanContext: Context

    fun onScanStart()

    fun onScanSuccess(data: List<AppModel>)

    fun onScanError(e: Throwable)
}
